pkg/hnsw/friends.go

```diff
@@ -0,0 +1,110 @@
+package hnsw
+
+import (
+	"errors"
+	"math"
+)
+
+type Point []float32
+
+type Friends struct {
+	friends []*BaseQueue
+}
```

Comment:
Since a API is helpful during the insertion and search logic, `Friends` serves as a struct over the priority queue.